superset/views/core.py

```diff
@@ -1846,25 +1847,33 @@ def dashboard(
         if not dashboard:
             abort(404)
 
-        if config["ENABLE_ACCESS_REQUEST"]:
-            for datasource in dashboard.datasources:
-                datasource = DatasourceDAO.get_datasource(
-                    datasource_type=DatasourceType(datasource.type),
-                    datasource_id=datasource.id,
-                    session=db.session(),
+        has_access_ = False
```

Comment:
@villebro can all this logic reside in the security manager? The issue is having security logic outside of the security manager adds complexity if you have custom logic in your security manager which grants everyone access to view the dashboard but block at the chart level if you don't have access to the underlying dataset.

Should lines 1851–1860-esque me added to the security manager which means this function would reduce to something of the form,

```python
try:
    security_manager.raise_for_dashboard_access(dashboard)
except DashboardAccessDeniedError:
    if config["ENABLE_ACCESS_REQUEST"]:
        flash(...)
        return redirect(f"/superset/request_access/?dashboard_id={dashboard.id}")
    else:
        flash(DashboardAccessDeniedError.message, "danger")
        return redirect(DASHBOARD_LIST_URL)
 ```

Reply:
I agree, it would probably make sense to have this in the security manager. However, in the scope of this PR I decided not to change the existing logic.